### ISSUE TITLE
WS-JMS A destination must be specified

### ISSUE THREAD

**Issue by Nyamiou** [seedstack/seed#128]
Hello,

I created a webservice that use SOAP over JMS and I want to be able to rollback a JMS message when the webservice throw an uncaught exception. It is working right now, but not in an elegant way  and it throw an exception :
```
[2015-09-24 17:44:23,067] ERROR [o.s.seed.ws.internal.jms.WSJmsMessageListener : 56] - An exception occurred during WS JMS message handling
java.lang.UnsupportedOperationException: A destination must be specified.
	at org.apache.activemq.ActiveMQMessageProducer.send(ActiveMQMessageProducer.java:245) ~[activemq-core-5.7.0.jar:5.7.0]
	at org.apache.activemq.ActiveMQMessageProducer.send(ActiveMQMessageProducer.java:212) ~[activemq-core-5.7.0.jar:5.7.0]
	at org.apache.activemq.ActiveMQMessageProducerSupport.send(ActiveMQMessageProducerSupport.java:241) ~[activemq-core-5.7.0.jar:5.7.0]
	at org.seedstack.seed.ws.internal.jms.JmsServerTransport.flush(JmsServerTransport.java:153) ~[seed-ws-support-jms-2.0.0-M3.jar:na]
	at org.seedstack.seed.ws.internal.jms.JmsAdapter.handle(JmsAdapter.java:39) ~[seed-ws-support-jms-2.0.0-M3.jar:na]
	at org.seedstack.seed.ws.internal.jms.WSJmsMessageListener.onMessage(WSJmsMessageListener.java:51) ~[seed-ws-support-jms-2.0.0-M3.jar:na]
	at org.seedstack.seed.jms.internal.MessageListenerAdapter.onMessage(MessageListenerAdapter.java:34) [seed-jms-support-core-2.0.0-M3.jar:na]
	at org.seedstack.seed.jms.SimpleMessagePoller.run(SimpleMessagePoller.java:93) [seed-jms-support-core-2.0.0-M3.jar:na]
	at java.lang.Thread.run(Thread.java:722) [na:1.7.0_21]
```

I think this is linked to the fact that in WSEndpointImpl every RuntimeException is caught and converted in a SOAP Fault response, then in JMSToolkit it interpret this as a response and tell JmsServerTransport to reply but it can't since my application doesn't have a JMS response queue so this exception is thrown and this is what actually trigger the rollback. 

Maybe JmsServerTransport or JMSToolkit should check if the message is a fault (Message.isFault()) and if there is not a "replyTo" destination should throw an exception like "Uncaught exception during message processing".

Note : My webservice endpoint response type is "void".

**Comment by adrienlauer** [comment]
Thank you for reporting the issue. It seems that the spec is indeed saying that SOAP faults shouldn't be returned to the sender in a one-way exchange but may be stored. As you proposed, I think that logging the fault along with a warning message is enough.

I'm scheduling this fix for 2.1.0 in november. Do you have time to fix it yourself in a pull request ? Just asking, don't feel obliged.

Note that we are currently moving WS support to their own repository as they will be available as an add-on in the upcoming november version. As a result, this issue may be moved [here](https://github.com/seedstack/web-services-addon) soon.

**Comment by adrienlauer** [comment]
This issue was moved to seedstack/web-services-addon#1
